test/Core.Test/Services/OrganizationServiceTests.cs

```diff
@@ -502,13 +505,187 @@ public async Task DeleteUsers_LastOwner(OrganizationUser orgUser, SutProvider<Or
+            User user, OrganizationUser orgUserAnotherOrg, Policy policy, string key, SutProvider<OrganizationService> sutProvider)
+        {
+            var organizationUserRepository = sutProvider.GetDependency<IOrganizationUserRepository>();
+            var organizationRepository = sutProvider.GetDependency<IOrganizationRepository>();
+            var policyRepository = sutProvider.GetDependency<IPolicyRepository>();
+            var userRepository = sutProvider.GetDependency<IUserRepository>();
+            var userService = Substitute.For<IUserService>();
+
+            org.PlanType = PlanType.EnterpriseAnnually;
+            orgUser.Status = OrganizationUserStatusType.Accepted;
+            orgUser.OrganizationId = confirmingUser.OrganizationId = org.Id;
+            orgUser.UserId = orgUserAnotherOrg.UserId = user.Id;
+            policy.Type = PolicyType.TwoFactorAuthentication;
+            policy.Enabled = true;
+            organizationUserRepository.GetManyAsync(default).ReturnsForAnyArgs(new[] {orgUser});
+            organizationUserRepository.GetManyByManyUsersAsync(default).ReturnsForAnyArgs(new[] {orgUserAnotherOrg});
+            organizationRepository.GetByIdAsync(org.Id).Returns(org);
+            userRepository.GetManyAsync(default).ReturnsForAnyArgs(new[] {user});
+            policyRepository.GetManyByOrganizationIdAsync(org.Id).Returns(new[] {policy});
+
+            var exception = await Assert.ThrowsAsync<BadRequestException>(
+                () => sutProvider.Sut.ConfirmUserAsync(orgUser.OrganizationId, orgUser.Id, key, confirmingUser.Id, userService));
+            Assert.Contains("User does not have two-step login enabled.", exception.Message);
+        }
+        
+        [Theory, CustomAutoData(typeof(SutProviderCustomization))]
+        public async Task ConfirmUser_Success(Organization org, OrganizationUser confirmingUser, OrganizationUser orgUser,
+            User user, Policy twoFactorPolicy, Policy singleOrgPolicy, string key, SutProvider<OrganizationService> sutProvider)
+        {
+            var organizationUserRepository = sutProvider.GetDependency<IOrganizationUserRepository>();
+            var organizationRepository = sutProvider.GetDependency<IOrganizationRepository>();
+            var policyRepository = sutProvider.GetDependency<IPolicyRepository>();
+            var userRepository = sutProvider.GetDependency<IUserRepository>();
+            var userService = Substitute.For<IUserService>();
+
+            org.PlanType = PlanType.EnterpriseAnnually;
+            orgUser.Status = OrganizationUserStatusType.Accepted;
+            orgUser.OrganizationId = confirmingUser.OrganizationId = org.Id;
+            orgUser.UserId = user.Id;
+            twoFactorPolicy.Type = PolicyType.TwoFactorAuthentication;
+            twoFactorPolicy.Enabled = true;
+            singleOrgPolicy.Type = PolicyType.SingleOrg;
+            singleOrgPolicy.Enabled = true;
```

Comment:
This test looks ripe for a fixture customization attribute to abstract as much hard coded data manipulation out of the test as possible. `PlanType`, `orgUser.Status`, etc. are used in several tests, so I bet there are some lines that could get deleted from other tests too by creating this.

I see that there is a lot of data in these tests that is dependent on other parameters (`ou.UserId = user.Id` type stuff). There's not a pattern set up to implement these in our fixtures yet, and those should continue to be set in the test like you already have it.

Reply:
Added a few fixtures which should clean things up.